apps/music/js/view.js

```diff
@@ -93,11 +93,41 @@ function View() {
   });
 }
 
-View.prototype.once = function(el, name, callback) {
-  el.addEventListener(name, function fn() {
-    el.removeEventListener(name, fn);
-    callback();
+View.prototype.setupSearch = function(params) {
+  this.searchBox.addEventListener('search', (evt) => this.search(evt.detail));
+
+  this.searchResults.addEventListener('open', () => {
+    this.client.method('searchOpen');
+  });
+
+  this.searchResults.addEventListener('close', () => {
+    this.client.method('searchClose');
+    this.list.scrollTop = this.searchBox.HEIGHT;
```

Comment:
I like that you're moving all the search setup into the base class. Might as well move the `HEIGHT` const here too so this doesn't accidentally get set to `undefined`.